### DISCUSSION TITLE
How to deploy Citrinet on android?

### DISCUSSION THREAD

**Discussion by amoghatr** [NVIDIA-NeMo/NeMo#4124] [Q&A]
Hi all, I've been trying to create a streaming asr app using a citrinet model. The pytorch demo app doesn't work with the citrinet model and there's not much documentation to refer to. Can someone point me to a tutorial or a repo? Thanks!

**Comment by titu1994** [comment]
We recommend Nvidia Riva for streaming purposes. While we do hand notebooks, they are for buffered inference, don't work with online audio as well, and are slow. 

If you just want buffered audio inference, you can use the scripts in the ASR examples directory : https://github.com/NVIDIA/NeMo/tree/main/examples/asr/asr_chunked_inference

**Comment by titu1994** [comment]
We are looking into alternative approaches to streaming in Nemo itself, but streaming inference is tough to do properly and efficiently so we still do recommend Riva for actual deployments.
